bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py

```diff
@@ -262,6 +267,291 @@ def predict(self):
+
+            # Collecting object detection results
+            objects = list()
+            # Create barrier
+            if self._exec_net.requests[request_id].wait(-1) == 0:
+                # Get output
+                output = self._exec_net.requests[request_id].outputs
+                # Iterate over output layers
+                for layer_name, out_blob in output.items():
+                    # Reshape output layer
+                    out_blob = out_blob.reshape(self._net.layers[self._net.layers[layer_name].parents[0]].shape)
+                    # Create layer params object
+                    layer_params = YoloParams(self._net.layers[layer_name].params, out_blob.shape[2])
+                    # Parse yolo bounding boxes out of output blob
+                    objects.extend(
+                        self._parse_yolo_region(
+                            out_blob,
+                            in_frame.shape[2:],
+                            self._image.shape[:-1],
+                            layer_params,
+                            self._prob_threshold))
+
+            # Filtering overlapping boxes
+            objects = sorted(objects, key=lambda obj : obj['confidence'], reverse=True)
+            for i in range(len(objects)):
+                if objects[i]['confidence'] == 0:
+                    continue
+                for j in range(i + 1, len(objects)):
+                    if self._intersection_over_union(objects[i], objects[j]) > self._iou_threshold:
+                        objects[j]['confidence'] = 0
+
+            # Convert objects to candidates
+            for yolo_object in objects:
+                # Drop out mostly 0 confidence objects
+                if yolo_object['confidence'] > self._prob_threshold:
+                    # Sort detections in classes
+                    if yolo_object['class_id'] == 0:
+                        self._ball_candidates.append(
+                            self._scaled_bbox_to_candidate(yolo_object)
+                        )
+                    if yolo_object['class_id'] == 1:
+                        self._goalpost_candidates.append(
+                            self._scaled_bbox_to_candidate(yolo_object)
+                        )
+
+
+class YoloParams:
+        """
+        Class to store params of yolo layers
+        """
+        def __init__(self, param, side):
+            self.num = 3 if 'num' not in param else int(param['num'])
+            self.coords = 4 if 'coords' not in param else int(param['coords'])
+            self.classes = 2 if 'classes' not in param else int(param['classes'])
+            self.anchors = [10.0, 13.0, 16.0, 30.0, 33.0, 23.0, 30.0, 61.0, 62.0, 45.0, 59.0, 119.0, 116.0, 90.0, 156.0,
+                            198.0,
+                            373.0, 326.0] if 'anchors' not in param else [float(a) for a in param['anchors'].split(',')]
+
+            if 'mask' in param:
+                mask = [int(idx) for idx in param['mask'].split(',')]
```

Comment:
What does idx stand for?

Reply:
I dont know exactly what this does, this is some yolo stuff.